planner/core/logical_plan_builder.go

```diff
@@ -987,7 +987,10 @@ func (b *PlanBuilder) buildSelection(ctx context.Context, p LogicalPlan, where a
 		for _, item := range cnfItems {
 			if con, ok := item.(*expression.Constant); ok && con.DeferredExpr == nil && con.ParamMarker == nil {
 				ret, _, err := expression.EvalBool(b.ctx, expression.CNFExprs{con}, chunk.Row{})
-				if err != nil || ret {
+				if err != nil {
+					return nil, errors.Trace(err)
```

Comment:
return error here may cause problems
```sql
update t set a = 'b' where ' ';
```
in tidb, it returns error, in MySQL it is ok.

Reply:
@zhaoxugang PTAL

Reply:
+1

Reply:
IMO the main question is that when convert err is present we should return err or return empty set？if return err maybe effect the user of tidb if return empty will confuse with actual return empty set.I need some advices
@xiongjiwei @yuqi1129 @morgo